rc/lsp.kak

```diff
@@ -1346,15 +1330,6 @@ method   = \"eclipse.jdt.ls/organizeImports\"
 " | eval "${kak_opt_lsp_cmd} --request") > /dev/null 2>&1 < /dev/null & }
 }
 
-# rust-analyzer extensions
-
-define-command -hidden rust-analyzer-inlay-hints -docstring "rust-analyzer-inlay-hints: request inlay hints (rust-analyzer).
-
-Deprecated: Delegates to lsp-experimental-inlay-hints. Once rust-analyzer switches to the official 
-textDocument/inlayHints request, this will be removed." %{
-    lsp-experimental-inlay-hints
```

Comment:
I think we should keep this compatibility wrapper because `rust-analyzer-inlay-hints` is used by [a bunch of people](https://github.com/search?q=rust-analyzer-inlay-hints&type=code). I'd like to avoid breaking old configurations unless there's a good reason. 
(I guess things will still break for people who don't update rust-analyzer and kak-lsp in lockstep but that's not our job.)
Of course the new `lsp-inlay-hints-enable` is nicer to use.

We have the convention of starting docstrings of such commands with DEPRECATED (and moving them to the bottom of the file I guess), so how about 


```kak
define-command -hidden rust-analyzer-inlay-hints -docstring "DEPRECATED, use lsp-inlay-hints-enable. request inlay hints" %{
    lsp-inlay-hints
}
```

Reply:
I can make a release this weekend, so we are compatible with current rust-analyzer

Reply:
I've pushed my suggestions, intend to merge like this. Dunno if a breaking change would be a big deal

	$ git range-diff origin/remove-experimental-inlay-hints...remove-experimental-inlay-hints
	 1:  cb43dd69 ! 27:  4a475dba remove rust-analyzer inlay-hints implementation
	    @@ Commit message
	     
	         rust-analyzer recently switched to the standard implementation
	     
	    +    Co-authored-by: Johannes Altmanninger <aclopte@gmail.com>
	    +
	      ## CHANGELOG.md ##
	     @@
	      ## Unreleased
	      
	    -+
	    -+Breaking changes
	    -+- Remove the old rust-analyzer-inlay-hints api. Inlay hints now requires rust-analyzer version >= 2022-04-18. See the README for how to set up the new inlay hints.
	    ++Deprecations:
	    ++- `rust-analyzer-inlay-hints` is deprecated in favor of `lsp-experimental-inlay-hints-enable`. Inlay hints now requires `rust-analyzer` version >= 2022-04-18. (#602, #613).
	     +
	      Additions:
	     -- Support the new `textDocument/inlayHint` request from LSP v3.17, and the same protocol with the `experimental/inlayHints` request currently used by rust-analyzer.
	    -+- Support the new `textDocument/inlayHint` request from LSP v3.17
	    ++- Support the new `textDocument/inlayHint` request from LSP v3.17.
	      
	      ## 12.1.0 - 2022-03-28
	      
	    @@ rc/lsp.kak: buf_line_count = ${kak_buf_line_count}
	      
	      define-command ccls-navigate -docstring "Navigate C/C++/ObjectiveC file" -params 1 %{
	     @@ rc/lsp.kak: method   = \"eclipse.jdt.ls/organizeImports\"
	    - " | eval "${kak_opt_lsp_cmd} --request") > /dev/null 2>&1 < /dev/null & }
	    - }
	      
	    --# rust-analyzer extensions
	    --
	    + # rust-analyzer extensions
	    + 
	     -define-command -hidden rust-analyzer-inlay-hints -docstring "rust-analyzer-inlay-hints: request inlay hints (rust-analyzer).
	     -
	     -Deprecated: Delegates to lsp-experimental-inlay-hints. Once rust-analyzer switches to the official 
	     -textDocument/inlayHints request, this will be removed." %{
	     -    lsp-experimental-inlay-hints
	    --}
	    --
	    ++define-command -hidden rust-analyzer-inlay-hints -docstring "DEPRECATED, use lsp-inlay-hints-enable. request inlay hints" %{
	    ++    lsp-inlay-hints
	    + }
	    + 
	      # texlab extensions
	    +@@ rc/lsp.kak: define-command lsp-inlay-hints-disable -params 1 -docstring "lsp-inlay-hints-dis
	    +     remove-hooks %arg{1} lsp-inlay-hints
	    + } -shell-script-candidates %{ printf '%s\n' buffer global window }
	    + 
	    +-define-command lsp-experimental-inlay-hints-enable -params 1 -docstring "lsp-experimental-inlay-hints-enable <scope>: enable inlay hints with experimental request for <scope>" %{
	    +-    add-highlighter "%arg{1}/lsp_inlay_hints" replace-ranges lsp_inlay_hints
	    +-    hook -group lsp-experimental-inlay-hints %arg{1} BufReload .* lsp-experimental-inlay-hints
	    +-    hook -group lsp-experimental-inlay-hints %arg{1} NormalIdle .* lsp-experimental-inlay-hints
	    +-    hook -group lsp-experimental-inlay-hints %arg{1} InsertIdle .* lsp-experimental-inlay-hints
	    +-} -shell-script-candidates %{ printf '%s\n' buffer global window }
	    +-
	    +-define-command lsp-experimental-inlay-hints-disable -params 1 -docstring "lsp-experimental-inlay-hints-disable <scope>: disable inlay hints with experimental request for <scope>"  %{
	    +-    remove-highlighter "%arg{1}/lsp_inlay_hints"
	    +-    remove-hooks %arg{1} lsp-experimental-inlay-hints
	    +-} -shell-script-candidates %{ printf '%s\n' buffer global window }
	    +-
	    + ### User mode ###
	      
	    - define-command texlab-forward-search -docstring "Request SyncTeX Forward Search for current line from the texlab language server
	    + declare-user-mode lsp
	     
	      ## src/controller.rs ##
	     @@ src/controller.rs: fn dispatch_editor_request(request: EditorRequest, ctx: &mut Context) {

Reply:
Looks good, thanks for taking over!